src/traces/heatmap/hover.js

```diff
@@ -96,8 +96,8 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode, contour)
     if(zmask && !zmask[ny][nx]) zVal = undefined;
 
     var text;
-    if(Array.isArray(trace.text) && Array.isArray(trace.text[ny])) {
-        text = trace.text[ny][nx];
+    if(Array.isArray(cd0.text) && Array.isArray(cd0.text[ny])) {
+        text = cd0.text[ny][nx];
```

Comment:
This looks fine as a bugfix, though I would think the more robust solution would not alter `fullData` at all, and *only* keep the re-dimensioned text array in `calcdata`.